api/app/Providers/GraphQLServiceProvider.php

```diff
@@ -139,642 +33,62 @@ static function (): EnumType {
-            }
-        );
+                    return new EnumType([
+                        'name' => $name,
+                        'values' => $values,
+                    ]);
+                }
+            );
+        }
 
-        $typeRegistry->registerLazy(
-            'OverallAssessmentStatus',
-            static function (): EnumType {
-                return new EnumType([
-                    'name' => 'OverallAssessmentStatus',
-                    'values' => array_column(OverallAssessmentStatus::cases(), 'name'),
-                ]);
-            }
-        );
+        $localizedEnums = Discover::in(app_path('Enums'))
+            ->enums()
+            ->custom(function (DiscoveredStructure $structure) {
+                return in_array(HasLocalization::class, class_uses($structure->getFcqn()));
+            })
+            ->sortBy(Sort::Name)
+            ->get();
 
-        foreach ($this->localizedEnums as $enum) {
+        foreach ($localizedEnums as $enum) {
             $name = class_basename($enum);
 
-            if (method_exists($enum, 'localizedString')) {
-                $typeRegistry->register(
-                    new ObjectType([
-                        'name' => 'Localized'.$name,
-                        'fields' => function () use ($typeRegistry, $name): array {
-                            return [
-                                'value' => Type::nonNull($typeRegistry->get($name)),
-                                'label' => Type::nonNull($typeRegistry->get('LocalizedString')),
-                            ];
-                        },
-                        /* @intelephense-ignore-next-line */
-                        'resolveField' => function ($value, array $args, $context, ResolveInfo $info) use ($enum) {
-                            switch ($info->fieldName) {
-                                case 'value': return $value;
-                                case 'label': return $enum::localizedString($value);
-                                default: return null;
-                            }
+            $typeRegistry->register(
+                new ObjectType([
+                    'name' => 'Localized'.$name,
+                    'fields' => function () use ($typeRegistry, $name): array {
+                        return [
+                            'value' => Type::nonNull($typeRegistry->get($name)),
+                            'label' => Type::nonNull($typeRegistry->get('LocalizedString')),
+                        ];
+                    },
+                    /** @disregard P1003 */
+                    'resolveField' => function ($value, array $args, $context, ResolveInfo $info) use ($enum) {
+                        switch ($info->fieldName) {
+                            case 'value': return $value;
+                                /** @disregard P1013, these enums do have the trait */
+                            case 'label': return $enum::localizedString($value);
```

Comment:
Similar here, I wonder if there is a way to typehint this? I had a `method_exists` check and even then it was still not happy :confounded: